src/Hakyll/Web/Template/Internal.hs

```diff
@@ -157,11 +158,14 @@ applyTemplate' tes name context x = go tes `catchError` handler
 
     applyElem Escaped = return "$"
 
-    applyElem (If e t mf) = (applyExpr e >> go t) `catchError` handler
+    applyElem (If e t mf) = do
+        c <- (applyExpr e >> return True) `compilerCatch` handler
+        if c
+            then go t
+            else maybe (return "") go mf
       where
-        handler _ = case mf of
-            Nothing -> return ""
-            Just f  -> go f
+        handler Error es = compilerThrow es
```

Comment:
To match the old behaviour, this would need to be
```hs
        handler Error es = do 
            logger <- compilerLogger <$> compilerAsk
            forM_ es $ \e -> compilerUnsafeIO $ Logger.debug logger $
                "Hakyll.Web.Template.applyTemplate: Error in 'if': " ++ e
            return ""
```